aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -361,65 +363,61 @@ private boolean isReadRestrictingOwner(AuthRule authRule) {
 
     private boolean isReadRestrictingStaticGroup(AuthRule authRule) {
         return AuthStrategy.GROUPS.equals(authRule.getAuthStrategy())
-            && authRule.getGroups() != null && !authRule.getGroups().isEmpty()
+            && !authRule.getGroups().isEmpty()
             && authRule.getOperationsOrDefault().contains(ModelOperation.READ);
     }
 
     private String getIdentityValue(String identityClaim, AuthorizationType authType) throws ApiException {
-        String identityValue = null;
-
         try {
-            identityValue = CognitoJWTParser
+            return CognitoJWTParser
                     .getPayload(getAuthToken(authType))
                     .getString(identityClaim);
         } catch (JSONException | CognitoParameterInvalidException error) {
-            // Could not read identity value from the token...
-            // Exception will be thrown so do nothing for now
-        }
-
-        if (identityValue == null || identityValue.isEmpty()) {
```

Comment:
The reason this was done the way it was is because an error won't be thrown if the identityClaim key is present but empty. And it seemed to make sense to throw the error in that case.

Reply:
i don't think that's something that should be caught in the client side. It should be treated the same as having an incorrect/non-existent username, which should be caught by the server.

Reply:
Fair enough